driver-core/src/main/com/mongodb/internal/connection/ByteBufferBsonOutput.java

```diff
@@ -91,13 +92,15 @@ public void writeByte(final int value) {
     }
 
     private ByteBuf getCurrentByteBuffer() {
-        ByteBuf curByteBuffer = getByteBufferAtIndex(curBufferIndex);
-        if (curByteBuffer.hasRemaining()) {
-            return curByteBuffer;
+        if (currentByteBuffer == null) {
+            currentByteBuffer = getByteBufferAtIndex(curBufferIndex);
+        }
+        if (currentByteBuffer.hasRemaining()) {
+            return currentByteBuffer;
         }
 
-        curBufferIndex++;
-        return getByteBufferAtIndex(curBufferIndex);
+        currentByteBuffer = getByteBufferAtIndex(++curBufferIndex);
+        return currentByteBuffer;
```

Comment:
```suggestion
        curBufferIndex++;
        currentByteBuffer = getByteBufferAtIndex(curBufferIndex);
        return currentByteBuffer;
```
Let's not use the value of the `++`/`--` operators. It is not incorrect, but this can be confusing.